### ISSUE TITLE
download_from() incorrectly handles gzipped response

### ISSUE THREAD

**Issue by pwmarcz** [ezhov-evgeny/webdav-client-python-3#61]
To reproduce, try creating an empty file on a server that supports gzip encoding, then download it using `download_from`:

```python
from io import BytesIO
buf = BytesIO()
client.download_from(buf, 'http://example.com/empty-file')
print(buf.getvalue())
```

Instead of empty content (`b''`), I'm getting `b'\x1f\x8b\x08\x00\x00\x00\x00\x00\x00\x03\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00'`, which is gzipped empty content.

It looks like the method uses `response.raw`, which does not decode gzip. A demonstration (https://httpbin.org/gzip should reply with a JSON body):

```python
>>> import requests
>>> response = requests.get('https://httpbin.org/gzip', stream=True)
>>> response.raw.read()
b'\x1f\x8b\x08\x00\xf7\xa7\xa1^\x02\xff= ...
```

**Comment by ezhov-evgeny** [comment]
Right it is described in requests docs https://requests.readthedocs.io/en/master/user/quickstart/#raw-response-content

I need a time for reproducing it on testing environment and then I'll be able to fix it.

**Comment by ezhov-evgeny** [comment]
@pwmarcz It is fixed in version 3.14.4
